aspnet/security/accconfirm.rst

```diff
@@ -1,236 +1,209 @@
 
+- Click the link to confirm your email.
 - Login with your email and password.
-
 - Log Off.
 
+Test password reset
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
 - Click Login and select **Forgot your password?**
+- Enter your email which was used to register the account with.
+- An email with the link to reset your password will be sent. Check your email and click it to reset your password.  After your password has been successfully reset, you can login with your email and new password.
+         
+Require email confirmation before login
+----------------------------------------
 
-.. image:: accconfirm/_static/loginaccconfirm3.PNG
+With the current templates, once a user completes the registration form, they are logged in. You generally want to confirm their email before logging them in. In the section below, we will modify the code to require new users have a confirmed email before they are logged in (authenticated). Update the ``HttpPost Login`` action with the following highlighted changes.
 
-- Enter your email which was used to register the account with.
+.. literalinclude:: accconfirm/sample/WebApplication1/src/WebApplication1/Controllers/AccountController.cs
+   :language: c#
+   :lines: 51-96
+   :emphasize-lines: 12-21
+   :dedent: 8
+
+- A security best practice is to not use production secrets in test and development. If you publish the app to Azure, you can set the SendGrid secrets as application settings in the Azure Web App portal. The configuration system is setup to read keys from environment variables.
 
-.. image:: accconfirm/_static/loginaccconfirm4.PNG
+Combine social and local login accounts
+-------------------------------------------
 
-- An email with the link to reset your password will be sent. Check your email and click it to reset your password.
+To complete this section, you must first enable an external authentication provider. See :doc:`sociallogins`.
 
-.. image:: accconfirm/_static/loginaccconfirm5.PNG
+You can combine local and social accounts by clicking on your email link. In the following sequence "RickAndMSFT@gmail.com" is first created as a local login, but you can create the account as a social log in first, then add a local login.
 
-- After your password has been successfully reset, you can login with your email and new password.
+.. image:: accconfirm/_static/rick.png
 
-.. image:: accconfirm/_static/loginaccconfirm6.PNG
+Click on the **Manage** link. Note the 0 external (social logins) associated with this account.
 
-         
-Require email confirmation before login
----------------------------------------
-Currently once a user completes the registration form, they are logged in. You generally want to confirm their email before logging them in. In the section below, we will modify the code to require new users to have a confirmed email before they are logged in (authenticated). Update the HttpPost Login action with the following highlighted changes.
```

Comment:
You used brackets previously when referring to an attributed version of an action method:
HttpPost Login > [HttpPost] Login